File: src/main/java/com/codeperfection/shipit/dto/TransporterDto.java
package com.codeperfection.shipit.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TransporterDto {

    private UUID uuid;

    @NotBlank
    private String name;

    @NotNull
    @Min(0)
    private Integer capacity;
}

File: src/main/kotlin/com/codeperfection/shipit/dto/transporter/TransporterDto.kt
package com.codeperfection.shipit.dto.transporter

import com.codeperfection.shipit.entity.Transporter
import java.util.*

data class TransporterDto(
    val id: UUID,
    val userId: UUID,
    val name: String,
    val capacity: Int
) {

    companion object {

        fun fromEntity(transporter: Transporter) = TransporterDto(
            id = transporter.id,
            userId = transporter.userId,
            name = transporter.name,
            capacity = transporter.capacity
        )
    }
}
